src/main/java/com/javadiscord/javabot/events/AutoMod.java

```diff
@@ -7,124 +7,140 @@
 import net.dv8tion.jda.api.EmbedBuilder;
 import net.dv8tion.jda.api.Permission;
 import net.dv8tion.jda.api.entities.Member;
-import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
 import net.dv8tion.jda.api.entities.Role;
 import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
 import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
 
 import java.awt.*;
 import java.util.Date;
-import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
-import java.util.stream.Collectors;
 
 public class AutoMod extends ListenerAdapter {
 
     private static final Pattern inviteURL = Pattern.compile("discord(?:(\\.(?:me|io|gg)|sites\\.com)/.{0,4}|app\\.com.{1,4}(?:invite|oauth2).{0,5}/)\\w+");
 
+    @Override
+    public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
+        try {
+            if (event.getMember().getUser().isBot() || event.getMember() == null)
+                return;
+        } catch (NullPointerException ignored) {
+            return;
+        }
+
+        if (event.getMember().hasPermission(Permission.MESSAGE_MANAGE))
+            return;
+
+        Member member = event.getMember();
+
+        // mention spam
+        if (event.getMessage().getMentionedMembers().size() >= 5) {
+            warn(event, member, "Automod: Mention Spam");
+        }
+
+        // Advertising
+        Matcher matcher = inviteURL.matcher(cleanString(event.getMessage().getContentRaw()));
+        if (matcher.find()) {
+            warn(event, member, "Automod: Advertising");
+        }
+
+        // spam
+        int spamCount = (int) event.getChannel().getIterableHistory().complete().stream()
+                .limit(10).filter(msg -> !msg.equals(event.getMessage()))
+                        // filter for spam
+                .filter(message -> message.getAuthor().equals(event.getAuthor()) && !message.getAuthor().isBot())
+                .filter(msg -> (event.getMessage().getTimeCreated().toEpochSecond() - msg.getTimeCreated().toEpochSecond()) < 6).count();
```

Comment:
This should not be done with a blocking `complete()` call... since this is done possibly for every guild message received.